docs/getting_started/self_host.md

```diff
@@ -0,0 +1,243 @@
+We will use [Traefik][traefik] as reverse proxy.  
+
+At the end of this tutorial, you will have a full-fledged version of Windmill,
+with LSP and a Postgres database running on your own server. We will deploy 
+with automatic LetsEncrypt encryption, handled by Traefik, using sensible
+defaults. 
+
+All the files required to achieve that are provided at the end of this
+tutorial in the [Example files](#example-files) section. These files are a
+good **starting point** for a internet-facing deployment of Windmill,
+and should be used as such. Without configuration they will not work and
+are not suitable for production deployments.
+
+## Intro
+
+To follow this tutorial, you will need Docker, `docker-compose` and a clone
+of the main [Windmill repository][windmill]:  
+
+`$ git clone git@github.com:windmill-labs/windmill.git`
+
+
+The repository comes with a minimal `docker-compose` file, without the 
+support for LSP, but with a Postgres database. It is suitable for 
+local deployments, and for discovering of the product. 
+
+If you want to deploy Windmill in a production setting, in a 
+internet facing environment, a few extra things are required.
+
+Below you will find a condensed list of high-level steps needed to get
+Windmill up and running. We will drill more into details in the 
+later parts of this tutorial.
+
+### Checklist
+
+Self-host pre-flight checklist for deployment using Traefik with TLS, 
+using ports `80` and `443` for HTTP and HTTPS respectively:
+
+Configuration: 
+
+- [ ] Did I set a strong `DB_PASSWORD` in the `.env` file?
+- [ ] Did I set the `BASE_URL` variable for `windmill` service in the 
+ `docker-compose.yml` file?
+- [ ] Did I add the lsp service to the `docker-compose.yml` file?
+- [ ] Did I add a reverse proxy service (e.g. Traefik) to the 
+  `docker-compose.yml` file?
+- [ ] Is the `windmill` service exposing a port different than `80` ?
+
+Running: 
+
+- [ ] Is the Traefik gateway running?
+- [ ] Is the lsp server running?
+- [ ] Did I route the `/ws` traffic to the lsp server?
+- [ ] Is my database hidden from the internet?
+
+## LSP Editor Support
+
+To add the LSP server to Windmill, and enable the formatting and intellisense 
+features in the editor, we will use the Windmill Language Server.
+
+The LSP communicates with websockets, is prefixed by `/ws/`, 
+i.e. `https://windmill.example/ws/` in the routing structure.
+
+To enable LSP you need to add the following service to the `docker-compose.yml` file: 
+
+```yaml
+lsp:
+  build: ./lsp
+  ports:
+    - 3001:3001
+```
+
+If you start the docker stack at this point, the LSP server will run. The support 
+in the editor however will not work, because it requires a reverse proxy to 
+handle path routing. If you've ran Windmill in development mode before, 
+that role was fulfilled by [Caddy][caddy].
+
+## Reverse proxy
+
+Windmill requires to be reverse-proxied at the `/ws` api prefix. In this example
```

Comment:
curious, why Traefik vs caddy ? 

Seems a bit inconsistent to have a dev setup with caddy and self-host with Traefik. I have no opinions of one vs the other, mostly about being consistent